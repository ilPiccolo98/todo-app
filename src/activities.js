const activities = [
  {
    id: 1,
    name: "shopping",
    description: "buy some stuff",
    status: false,
  },
  {
    id: 2,
    name: "have fun with my dog",
    description: "take my dog for a walk and have fun with it",
    status: true,
  },
  {
    id: 3,
    name: "work out",
    description: "work out a little bit with weights and barbells",
    status: false,
  },
];

export default activities;
