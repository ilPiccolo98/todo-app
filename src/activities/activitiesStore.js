import { configureStore } from "@reduxjs/toolkit";
import activitiesSlice from "./activitiesSlice";

const storeActivities = configureStore({
  reducer: {
    activities: activitiesSlice,
  },
});

export default storeActivities;
