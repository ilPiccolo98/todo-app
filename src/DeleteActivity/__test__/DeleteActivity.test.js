/* eslint-disable import/first */
import React from "react";
jest.mock("../../activities/initialActivities", () => {
  return () => [
    {
      id: 1,
      name: "shopping",
      description: "buy some stuff",
      status: false,
    },
  ];
});
import { render, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Provider } from "react-redux";
import DeleteActivity from "../DeleteActivity";
import activitiesStore from "../../activities/activitiesStore";
import * as actions from "../../activities/activitiesReducer";

const renderDeleteActivity = () => {
  return render(
    <Provider store={activitiesStore}>
      <DeleteActivity />
    </Provider>
  );
};

const getDeleteActvityFormComponents = (component) => {
  const idFieldDeleteActivity = component.getByTestId(
    "id-field-delete-activity"
  );
  const deleteButtonDeleteActivity = component.getByTestId(
    "delete-button-delete-activity"
  );
  return { idFieldDeleteActivity, deleteButtonDeleteActivity };
};

describe("testing DeleteActivity component", () => {
  it("should call the deleteActivity action with id=1 after pressed button", async () => {
    const component = renderDeleteActivity();
    const { idFieldDeleteActivity, deleteButtonDeleteActivity } =
      getDeleteActvityFormComponents(component);
    const deleteActivitySpy = jest.spyOn(actions, "deleteActivity");
    await waitFor(() => {
      fireEvent.change(idFieldDeleteActivity, {
        target: {
          value: "1",
        },
      });
    });
    await waitFor(() => {
      fireEvent.click(deleteButtonDeleteActivity);
    });
    expect(deleteActivitySpy).toHaveBeenCalledWith("1");
    expect(component).toMatchSnapshot();
  });

  it("shouldn't call the deleteActivity action with id=10 after pressed the delete button", async () => {
    const component = renderDeleteActivity();
    const { idFieldDeleteActivity, deleteButtonDeleteActivity } =
      getDeleteActvityFormComponents(component);
    const deleteActivitySpy = jest.spyOn(actions, "deleteActivity");
    await waitFor(() => {
      fireEvent.change(idFieldDeleteActivity, {
        target: {
          value: "10",
        },
      });
    });
    await waitFor(() => {
      fireEvent.click(deleteButtonDeleteActivity);
    });
    expect(deleteActivitySpy).not.toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });
});
