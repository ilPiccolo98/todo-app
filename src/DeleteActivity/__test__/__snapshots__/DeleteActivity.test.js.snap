// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteActivity Component should let you delete an activity using the DeleteActivity component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table>
          <tbody
            data-testid="table-body"
          >
            <tr>
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
            <tr>
              <td
                data-testid="td-id-2"
              >
                2
              </td>
              <td
                data-testid="td-name-2"
              >
                have fun with my dog
              </td>
              <td
                data-testid="td-description-2"
              >
                take my dog for a walk and have fun with it
              </td>
              <td
                data-testid="td-status-2"
              >
                true
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-3"
              >
                3
              </td>
              <td
                data-testid="td-name-3"
              >
                work out
              </td>
              <td
                data-testid="td-description-3"
              >
                work out a little bit with weights and barbells
              </td>
              <td
                data-testid="td-status-3"
              >
                false
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="buttons-panel"
      >
        <a
          href="/add-activity"
        >
          <input
            data-testid="insert-button-app"
            type="button"
            value="Insert activity"
          />
        </a>
        <a
          href="/update-activity"
        >
          <input
            data-testid="update-button-app"
            type="button"
            value="Update activity"
          />
        </a>
        <a
          href="/delete-activity"
        >
          <input
            data-testid="delete-button-app"
            type="button"
            value="Delete activity"
          />
        </a>
      </div>
      <form
        action="#"
      >
        <div>
          <label
            for="id"
          >
            Id
          </label>
          <input
            data-testid="id-field-delete-activity"
            id="id"
            name="id"
            patterm="[0-9]*"
            placeholder="Id"
            type="text"
            value="1"
          />
        </div>
        <input
          data-testid="delete-button-delete-activity"
          type="submit"
          value="Delete"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <div>
      <table>
        <tbody
          data-testid="table-body"
        >
          <tr>
            <th>
              Id
            </th>
            <th>
              Name
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
          <tr>
            <td
              data-testid="td-id-2"
            >
              2
            </td>
            <td
              data-testid="td-name-2"
            >
              have fun with my dog
            </td>
            <td
              data-testid="td-description-2"
            >
              take my dog for a walk and have fun with it
            </td>
            <td
              data-testid="td-status-2"
            >
              true
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-3"
            >
              3
            </td>
            <td
              data-testid="td-name-3"
            >
              work out
            </td>
            <td
              data-testid="td-description-3"
            >
              work out a little bit with weights and barbells
            </td>
            <td
              data-testid="td-status-3"
            >
              false
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="buttons-panel"
    >
      <a
        href="/add-activity"
      >
        <input
          data-testid="insert-button-app"
          type="button"
          value="Insert activity"
        />
      </a>
      <a
        href="/update-activity"
      >
        <input
          data-testid="update-button-app"
          type="button"
          value="Update activity"
        />
      </a>
      <a
        href="/delete-activity"
      >
        <input
          data-testid="delete-button-app"
          type="button"
          value="Delete activity"
        />
      </a>
    </div>
    <form
      action="#"
    >
      <div>
        <label
          for="id"
        >
          Id
        </label>
        <input
          data-testid="id-field-delete-activity"
          id="id"
          name="id"
          patterm="[0-9]*"
          placeholder="Id"
          type="text"
          value="1"
        />
      </div>
      <input
        data-testid="delete-button-delete-activity"
        type="submit"
        value="Delete"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DeleteActivity Component should let you type data into the form after pressed the delete button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table>
          <tbody
            data-testid="table-body"
          >
            <tr>
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
            <tr>
              <td
                data-testid="td-id-1"
              >
                1
              </td>
              <td
                data-testid="td-name-1"
              >
                shopping
              </td>
              <td
                data-testid="td-description-1"
              >
                buy some stuff
              </td>
              <td
                data-testid="td-status-1"
              >
                false
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-2"
              >
                2
              </td>
              <td
                data-testid="td-name-2"
              >
                have fun with my dog
              </td>
              <td
                data-testid="td-description-2"
              >
                take my dog for a walk and have fun with it
              </td>
              <td
                data-testid="td-status-2"
              >
                true
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-3"
              >
                3
              </td>
              <td
                data-testid="td-name-3"
              >
                work out
              </td>
              <td
                data-testid="td-description-3"
              >
                work out a little bit with weights and barbells
              </td>
              <td
                data-testid="td-status-3"
              >
                false
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="buttons-panel"
      >
        <a
          href="/add-activity"
        >
          <input
            data-testid="insert-button-app"
            type="button"
            value="Insert activity"
          />
        </a>
        <a
          href="/update-activity"
        >
          <input
            data-testid="update-button-app"
            type="button"
            value="Update activity"
          />
        </a>
        <a
          href="/delete-activity"
        >
          <input
            data-testid="delete-button-app"
            type="button"
            value="Delete activity"
          />
        </a>
      </div>
      <form
        action="#"
      >
        <div>
          <label
            for="id"
          >
            Id
          </label>
          <input
            data-testid="id-field-delete-activity"
            id="id"
            name="id"
            patterm="[0-9]*"
            placeholder="Id"
            type="text"
            value="1"
          />
        </div>
        <input
          data-testid="delete-button-delete-activity"
          type="submit"
          value="Delete"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <div>
      <table>
        <tbody
          data-testid="table-body"
        >
          <tr>
            <th>
              Id
            </th>
            <th>
              Name
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
          <tr>
            <td
              data-testid="td-id-1"
            >
              1
            </td>
            <td
              data-testid="td-name-1"
            >
              shopping
            </td>
            <td
              data-testid="td-description-1"
            >
              buy some stuff
            </td>
            <td
              data-testid="td-status-1"
            >
              false
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-2"
            >
              2
            </td>
            <td
              data-testid="td-name-2"
            >
              have fun with my dog
            </td>
            <td
              data-testid="td-description-2"
            >
              take my dog for a walk and have fun with it
            </td>
            <td
              data-testid="td-status-2"
            >
              true
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-3"
            >
              3
            </td>
            <td
              data-testid="td-name-3"
            >
              work out
            </td>
            <td
              data-testid="td-description-3"
            >
              work out a little bit with weights and barbells
            </td>
            <td
              data-testid="td-status-3"
            >
              false
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="buttons-panel"
    >
      <a
        href="/add-activity"
      >
        <input
          data-testid="insert-button-app"
          type="button"
          value="Insert activity"
        />
      </a>
      <a
        href="/update-activity"
      >
        <input
          data-testid="update-button-app"
          type="button"
          value="Update activity"
        />
      </a>
      <a
        href="/delete-activity"
      >
        <input
          data-testid="delete-button-app"
          type="button"
          value="Delete activity"
        />
      </a>
    </div>
    <form
      action="#"
    >
      <div>
        <label
          for="id"
        >
          Id
        </label>
        <input
          data-testid="id-field-delete-activity"
          id="id"
          name="id"
          patterm="[0-9]*"
          placeholder="Id"
          type="text"
          value="1"
        />
      </div>
      <input
        data-testid="delete-button-delete-activity"
        type="submit"
        value="Delete"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DeleteActivity Component should show the DeleteActivity form empty after pressed the delete button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table>
          <tbody
            data-testid="table-body"
          >
            <tr>
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
            <tr>
              <td
                data-testid="td-id-1"
              >
                1
              </td>
              <td
                data-testid="td-name-1"
              >
                shopping
              </td>
              <td
                data-testid="td-description-1"
              >
                buy some stuff
              </td>
              <td
                data-testid="td-status-1"
              >
                false
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-2"
              >
                2
              </td>
              <td
                data-testid="td-name-2"
              >
                have fun with my dog
              </td>
              <td
                data-testid="td-description-2"
              >
                take my dog for a walk and have fun with it
              </td>
              <td
                data-testid="td-status-2"
              >
                true
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-3"
              >
                3
              </td>
              <td
                data-testid="td-name-3"
              >
                work out
              </td>
              <td
                data-testid="td-description-3"
              >
                work out a little bit with weights and barbells
              </td>
              <td
                data-testid="td-status-3"
              >
                false
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="buttons-panel"
      >
        <a
          href="/add-activity"
        >
          <input
            data-testid="insert-button-app"
            type="button"
            value="Insert activity"
          />
        </a>
        <a
          href="/update-activity"
        >
          <input
            data-testid="update-button-app"
            type="button"
            value="Update activity"
          />
        </a>
        <a
          href="/delete-activity"
        >
          <input
            data-testid="delete-button-app"
            type="button"
            value="Delete activity"
          />
        </a>
      </div>
      <form
        action="#"
      >
        <div>
          <label
            for="id"
          >
            Id
          </label>
          <input
            data-testid="id-field-delete-activity"
            id="id"
            name="id"
            patterm="[0-9]*"
            placeholder="Id"
            type="text"
            value=""
          />
        </div>
        <input
          data-testid="delete-button-delete-activity"
          type="submit"
          value="Delete"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <div>
      <table>
        <tbody
          data-testid="table-body"
        >
          <tr>
            <th>
              Id
            </th>
            <th>
              Name
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
          <tr>
            <td
              data-testid="td-id-1"
            >
              1
            </td>
            <td
              data-testid="td-name-1"
            >
              shopping
            </td>
            <td
              data-testid="td-description-1"
            >
              buy some stuff
            </td>
            <td
              data-testid="td-status-1"
            >
              false
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-2"
            >
              2
            </td>
            <td
              data-testid="td-name-2"
            >
              have fun with my dog
            </td>
            <td
              data-testid="td-description-2"
            >
              take my dog for a walk and have fun with it
            </td>
            <td
              data-testid="td-status-2"
            >
              true
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-3"
            >
              3
            </td>
            <td
              data-testid="td-name-3"
            >
              work out
            </td>
            <td
              data-testid="td-description-3"
            >
              work out a little bit with weights and barbells
            </td>
            <td
              data-testid="td-status-3"
            >
              false
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="buttons-panel"
    >
      <a
        href="/add-activity"
      >
        <input
          data-testid="insert-button-app"
          type="button"
          value="Insert activity"
        />
      </a>
      <a
        href="/update-activity"
      >
        <input
          data-testid="update-button-app"
          type="button"
          value="Update activity"
        />
      </a>
      <a
        href="/delete-activity"
      >
        <input
          data-testid="delete-button-app"
          type="button"
          value="Delete activity"
        />
      </a>
    </div>
    <form
      action="#"
    >
      <div>
        <label
          for="id"
        >
          Id
        </label>
        <input
          data-testid="id-field-delete-activity"
          id="id"
          name="id"
          patterm="[0-9]*"
          placeholder="Id"
          type="text"
          value=""
        />
      </div>
      <input
        data-testid="delete-button-delete-activity"
        type="submit"
        value="Delete"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DeleteActivity Component shouldn't delete a not existing activity 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table>
          <tbody
            data-testid="table-body"
          >
            <tr>
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
            <tr>
              <td
                data-testid="td-id-1"
              >
                1
              </td>
              <td
                data-testid="td-name-1"
              >
                shopping
              </td>
              <td
                data-testid="td-description-1"
              >
                buy some stuff
              </td>
              <td
                data-testid="td-status-1"
              >
                false
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-2"
              >
                2
              </td>
              <td
                data-testid="td-name-2"
              >
                have fun with my dog
              </td>
              <td
                data-testid="td-description-2"
              >
                take my dog for a walk and have fun with it
              </td>
              <td
                data-testid="td-status-2"
              >
                true
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-3"
              >
                3
              </td>
              <td
                data-testid="td-name-3"
              >
                work out
              </td>
              <td
                data-testid="td-description-3"
              >
                work out a little bit with weights and barbells
              </td>
              <td
                data-testid="td-status-3"
              >
                false
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="buttons-panel"
      >
        <a
          href="/add-activity"
        >
          <input
            data-testid="insert-button-app"
            type="button"
            value="Insert activity"
          />
        </a>
        <a
          href="/update-activity"
        >
          <input
            data-testid="update-button-app"
            type="button"
            value="Update activity"
          />
        </a>
        <a
          href="/delete-activity"
        >
          <input
            data-testid="delete-button-app"
            type="button"
            value="Delete activity"
          />
        </a>
      </div>
      <form
        action="#"
      >
        <div>
          <label
            for="id"
          >
            Id
          </label>
          <input
            data-testid="id-field-delete-activity"
            id="id"
            name="id"
            patterm="[0-9]*"
            placeholder="Id"
            type="text"
            value="20"
          />
        </div>
        <input
          data-testid="delete-button-delete-activity"
          type="submit"
          value="Delete"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <div>
      <table>
        <tbody
          data-testid="table-body"
        >
          <tr>
            <th>
              Id
            </th>
            <th>
              Name
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
          <tr>
            <td
              data-testid="td-id-1"
            >
              1
            </td>
            <td
              data-testid="td-name-1"
            >
              shopping
            </td>
            <td
              data-testid="td-description-1"
            >
              buy some stuff
            </td>
            <td
              data-testid="td-status-1"
            >
              false
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-2"
            >
              2
            </td>
            <td
              data-testid="td-name-2"
            >
              have fun with my dog
            </td>
            <td
              data-testid="td-description-2"
            >
              take my dog for a walk and have fun with it
            </td>
            <td
              data-testid="td-status-2"
            >
              true
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-3"
            >
              3
            </td>
            <td
              data-testid="td-name-3"
            >
              work out
            </td>
            <td
              data-testid="td-description-3"
            >
              work out a little bit with weights and barbells
            </td>
            <td
              data-testid="td-status-3"
            >
              false
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="buttons-panel"
    >
      <a
        href="/add-activity"
      >
        <input
          data-testid="insert-button-app"
          type="button"
          value="Insert activity"
        />
      </a>
      <a
        href="/update-activity"
      >
        <input
          data-testid="update-button-app"
          type="button"
          value="Update activity"
        />
      </a>
      <a
        href="/delete-activity"
      >
        <input
          data-testid="delete-button-app"
          type="button"
          value="Delete activity"
        />
      </a>
    </div>
    <form
      action="#"
    >
      <div>
        <label
          for="id"
        >
          Id
        </label>
        <input
          data-testid="id-field-delete-activity"
          id="id"
          name="id"
          patterm="[0-9]*"
          placeholder="Id"
          type="text"
          value="20"
        />
      </div>
      <input
        data-testid="delete-button-delete-activity"
        type="submit"
        value="Delete"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DeleteActivity Component shouldn't let you delete an activity with id field blank 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table>
          <tbody
            data-testid="table-body"
          >
            <tr>
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
            <tr>
              <td
                data-testid="td-id-1"
              >
                1
              </td>
              <td
                data-testid="td-name-1"
              >
                shopping
              </td>
              <td
                data-testid="td-description-1"
              >
                buy some stuff
              </td>
              <td
                data-testid="td-status-1"
              >
                false
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-2"
              >
                2
              </td>
              <td
                data-testid="td-name-2"
              >
                have fun with my dog
              </td>
              <td
                data-testid="td-description-2"
              >
                take my dog for a walk and have fun with it
              </td>
              <td
                data-testid="td-status-2"
              >
                true
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-3"
              >
                3
              </td>
              <td
                data-testid="td-name-3"
              >
                work out
              </td>
              <td
                data-testid="td-description-3"
              >
                work out a little bit with weights and barbells
              </td>
              <td
                data-testid="td-status-3"
              >
                false
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="buttons-panel"
      >
        <a
          href="/add-activity"
        >
          <input
            data-testid="insert-button-app"
            type="button"
            value="Insert activity"
          />
        </a>
        <a
          href="/update-activity"
        >
          <input
            data-testid="update-button-app"
            type="button"
            value="Update activity"
          />
        </a>
        <a
          href="/delete-activity"
        >
          <input
            data-testid="delete-button-app"
            type="button"
            value="Delete activity"
          />
        </a>
      </div>
      <form
        action="#"
      >
        <div>
          <label
            for="id"
          >
            Id
          </label>
          <input
            data-testid="id-field-delete-activity"
            id="id"
            name="id"
            patterm="[0-9]*"
            placeholder="Id"
            type="text"
            value=""
          />
        </div>
        <input
          data-testid="delete-button-delete-activity"
          type="submit"
          value="Delete"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <div>
      <table>
        <tbody
          data-testid="table-body"
        >
          <tr>
            <th>
              Id
            </th>
            <th>
              Name
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
          <tr>
            <td
              data-testid="td-id-1"
            >
              1
            </td>
            <td
              data-testid="td-name-1"
            >
              shopping
            </td>
            <td
              data-testid="td-description-1"
            >
              buy some stuff
            </td>
            <td
              data-testid="td-status-1"
            >
              false
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-2"
            >
              2
            </td>
            <td
              data-testid="td-name-2"
            >
              have fun with my dog
            </td>
            <td
              data-testid="td-description-2"
            >
              take my dog for a walk and have fun with it
            </td>
            <td
              data-testid="td-status-2"
            >
              true
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-3"
            >
              3
            </td>
            <td
              data-testid="td-name-3"
            >
              work out
            </td>
            <td
              data-testid="td-description-3"
            >
              work out a little bit with weights and barbells
            </td>
            <td
              data-testid="td-status-3"
            >
              false
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="buttons-panel"
    >
      <a
        href="/add-activity"
      >
        <input
          data-testid="insert-button-app"
          type="button"
          value="Insert activity"
        />
      </a>
      <a
        href="/update-activity"
      >
        <input
          data-testid="update-button-app"
          type="button"
          value="Update activity"
        />
      </a>
      <a
        href="/delete-activity"
      >
        <input
          data-testid="delete-button-app"
          type="button"
          value="Delete activity"
        />
      </a>
    </div>
    <form
      action="#"
    >
      <div>
        <label
          for="id"
        >
          Id
        </label>
        <input
          data-testid="id-field-delete-activity"
          id="id"
          name="id"
          patterm="[0-9]*"
          placeholder="Id"
          type="text"
          value=""
        />
      </div>
      <input
        data-testid="delete-button-delete-activity"
        type="submit"
        value="Delete"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
